version: '3.1'
services:
  db:
    container_name: psql_db
    image: postgres:12
    environment:
      POSTGRES_USER: jorgepl
      POSTGRES_PASSWORD: admin
      POSTGRES_DB: trivia_test
    ports:
      - 5677:5432
    volumes: 
      - ./backend/scripts:/docker-entrypoint-initdb.d
    networks:
      myapi:
        ipv4_address: "172.25.0.2"
    healthcheck:
      test: ["CMD", "pg_isready", "--dbname=trivia_test", "--username=jorgepl"]
      interval: 10s
      timeout: 5s
      retries: 5

  flask_api:
    container_name: flaskr
    build:
      context: ./backend
      dockerfile: ./flaskr/Dockerfile
    environment:
      FLASK_APP: flaskr
      FLASK_ENV: development
      PYTHONPATH: "/app"
    ports:
      - 5000:5000
    networks:
      myapi:
        ipv4_address: "172.25.0.3"
    depends_on:
      db:
        condition: service_healthy

  react-app: 
    container_name: trivia-frontend
    build:
      context: ./frontend
      dockerfile: ./Dockerfile
    ports:
      - 3000:3000
    networks:
      - myapi
    depends_on:
      - flask_api

networks:
  myapi:
    driver: bridge
    ipam:
      config:
        - subnet: 172.25.0.0/24

